### PR TITLE
Protobuf enhancements docs update

### PR DESCRIPTION
**Pull request by nmcdonnell-kx**

### CODE REVIEW

File: docs/interfaces/protobuf/reference.md [hunk 1]
```diff
@@ -154,66 +177,138 @@ q).protobufkdb.listImportedMessageTypes[]
...
 
 ```txt
-.protobufkdb.loadMessage[message_type;file_name]
+.protobufkdb.displayMessageSchema[message_type]
+```
+
+Where `message_type` is the name of a message type as a string, 
+
+1.  prints schema format to stdout 
+1.  returns generic null
```
Comment (mshimizu-kx) [inline docs/interfaces/protobuf/reference.md:191]: Need new line.

File: docs/interfaces/protobuf/reference.md [hunk 2]
```diff
@@ -154,66 +177,138 @@ q).protobufkdb.listImportedMessageTypes[]
...
+
+## `getMessageFields`
+
+_Get the list of message fields_
+
+```txt
+.protobufkdb.getMessageFields[message_type]
+```
+
+Where `message_type` is a message type (string or symbol) matching a message name in the `.proto` definition
```
Comment (mshimizu-kx) [inline docs/interfaces/protobuf/reference.md:216]: Itemize?
Ref) https://code.kx.com/q/interfaces/kafka/reference/#kfksubscription

File: docs/interfaces/protobuf/reference.md [hunk 3]
```diff
@@ -223,141 +318,323 @@ Where:
...
+
+_Parse from a Protobuf message file to a kdb+ mixed list object_
 
 ```txt
-.protobufkdb.version[]
+.protobufkdb.loadMessageToList[message_type;file_name]
 ```
 
-returns the version of the `libprotobuf` shared object used by the interface, as an integer.
+Where:
```
Comment (mshimizu-kx) [inline docs/interfaces/protobuf/reference.md]: Trim `:`

File: docs/interfaces/protobuf/reference.md [hunk 4]
```diff
@@ -223,141 +318,323 @@ Where:
...
+.protobufkdb.saveMessageFromDict[message_type;file_name;msg_in]
+```
+
+Where
+
+-   `message_type` is a message type (string or symbol) matching a message name in the `.proto` definition
+-   `file_name` is the name of a file (string or symbol)
+-   `msg_in` is a kdb+ dictionary from field name symbols to field values
+
+the function
```
Comment (mshimizu-kx) [inline docs/interfaces/protobuf/reference.md:531]: Need this line? For example loadMessage does not have this line.

File: docs/interfaces/protobuf/reference.md [hunk 5]
```diff
@@ -223,141 +318,323 @@ Where:
...
+```txt
+.protobufkdb.parseArrayDebug[message_type;char_array]
 ```
 
+Where
 
+-   `message_type` is a message type (string or symbol) matching a message name in the `.proto` definition
+-   `char_array` is the serialized Protobuf message (string)
+
+the function
```
Comment (mshimizu-kx) [inline docs/interfaces/protobuf/reference.md:584]: Need this line?

File: docs/interfaces/protobuf/reference.md [hunk 6]
```diff
@@ -223,141 +318,323 @@ Where:
...
+```txt
+.protobufkdb.loadMessageDebug[message_type;file_name]
+```
+
+Where:
+
+-   `message_type` is a message type (string or symbol) matching a message name in the `.proto` definition
+-   `file_name` is the name of a file (string or symbol)
+
+the function
```
Comment (mshimizu-kx) [inline docs/interfaces/protobuf/reference.md:618]: Need this line?

File: docs/interfaces/protobuf/reference.md [hunk 7]
```diff
@@ -223,141 +318,323 @@ Where:
...
+
+## `loadMessageDebug`
+
+_Parse from a Protobuf message file and display the debugging_
+
+```txt
+.protobufkdb.loadMessageDebug[message_type;file_name]
+```
+
+Where:
```
Comment (mshimizu-kx) [inline docs/interfaces/protobuf/reference.md]: Trim `:`

File: docs/interfaces/protobuf/reference.md [hunk 8]
```diff
@@ -223,141 +318,323 @@ Where:
...
+
+_Parse from a Protobuf message file to a kdb+ dictionary_
 
 ```txt
-.protobufkdb.versionStr[]
+.protobufkdb.loadMessageToDict[message_type;file_name]
 ```
 
-returns the version of the `libprotobuf` shared object used by the interface, as a string.
+Where:
```
Comment (mshimizu-kx) [inline docs/interfaces/protobuf/reference.md]: Trim `:`

**Review by mshimizu-kx — CHANGES_REQUESTED** [review]
- Some minor cosmetic
- Add some note about trailing `::` for a message whose all elements have the same type? Also mention the fact that the parsed result has type 0 but does not have `::`?

File: docs/interfaces/protobuf/reference.md (hunk 2 above)
Comment (nmcdonnell-kx) [reply docs/interfaces/protobuf/reference.md:216]: Don't think so (looking at current protobuf docs).  The function only takes a single arg so isn't itemised.

File: docs/interfaces/protobuf/reference.md (hunk 3 above)
Comment (nmcdonnell-kx) [reply docs/interfaces/protobuf/reference.md]: done

File: docs/interfaces/protobuf/reference.md (hunk 4 above)
Comment (nmcdonnell-kx) [reply docs/interfaces/protobuf/reference.md:531]: Again, I think 'the function' is used when the function's operations is described in multiple steps

File: docs/interfaces/protobuf/reference.md (hunk 5 above)
Comment (nmcdonnell-kx) [reply docs/interfaces/protobuf/reference.md:584]: ditto

File: docs/interfaces/protobuf/reference.md (hunk 6 above)
Comment (nmcdonnell-kx) [reply docs/interfaces/protobuf/reference.md:618]: ditto

File: docs/interfaces/protobuf/reference.md (hunk 7 above)
Comment (nmcdonnell-kx) [reply docs/interfaces/protobuf/reference.md]: done

File: docs/interfaces/protobuf/reference.md (hunk 8 above)
Comment (nmcdonnell-kx) [reply docs/interfaces/protobuf/reference.md]: done

**Comment by nmcdonnell-kx** [comment]
- Add some note about trailing :: for a message whose all elements have the same type? Also mention the fact that the parsed result has type 0 but does not have ::?

:: is described in protobuf-types.md (at the top).  I added a bit about parsing not returning ::

**Review by mshimizu-kx — APPROVED** [review]
Good

**Pull request merged**